cmd/plugin.go

```diff
@@ -277,51 +288,89 @@ func runPluginUpdateCmd(cmd *cobra.Command, args []string) {
 		return
 	}
 
-	// we can't allow update and install at the same time
+	if len(plugins) > 0 && cmdconfig.Viper().GetBool("all") {
+		// we can't allow update and install at the same time
+		fmt.Println()
+		utils.ShowError(fmt.Errorf("%s cannot be used when updating specific plugins", constants.Bold("`--all`")))
+		fmt.Println()
+		cmd.Help()
+		fmt.Println()
+		return
+	}
+
+	updateSkipped := []updateSkip{}
+
+	state, err := statefile.LoadState()
+	if err != nil {
+		utils.ShowError(fmt.Errorf("could not load state"))
+		return
+	}
+
+	// load up the version file data
+	versionData, err := versionfile.Load()
+	if err != nil {
+		utils.ShowError(fmt.Errorf("error loading current plugin data"))
+		return
+	}
+	if versionData.Plugins == nil {
+		versionData.Plugins = map[string]*versionfile.InstalledVersion{}
+	}
+
+	runUpdatesFor := []*versionfile.InstalledVersion{}
+
 	if cmdconfig.Viper().GetBool("all") {
-		if len(plugins) > 0 {
-			utils.ShowError(fmt.Errorf("%s cannot be used when updating specific plugins", constants.Bold("`--all`")))
-			return
-		}
+		for k, v := range versionData.Plugins {
+			ref := ociinstaller.NewSteampipeImageRef(k)
+			org, name, stream := ref.GetOrgNameAndStream()
+			key := fmt.Sprintf("%s/%s@%s", org, name, stream)
 
-		// get the update report
-		plugins = []string{}
-		state, _ := statefile.LoadState()
-		reports := pluginmanager.GetPluginUpdateReport(state.InstallationID)
-		for _, report := range reports {
-			if report.Plugin.ImageDigest != report.CheckResponse.Digest {
-				plugins = append(plugins, fmt.Sprintf("%s/%s@%s", report.CheckResponse.Org, report.CheckResponse.Name, report.CheckResponse.Stream))
+			plugins = append(plugins, key)
+			runUpdatesFor = append(runUpdatesFor, v)
+		}
+	} else {
+		// get the args and retrieve the installed versions
+		for _, p := range plugins {
+			ref := ociinstaller.NewSteampipeImageRef(p)
+			isExists, _ := pluginmanager.IsPluginExists(p)
```

Comment:
rename var to `exists`, `pluginmanager` package to `plugin` and  `IsPluginExists` to `Exists`, so:
```
exists, _ := plugin.Exists(p)
```